crates/kitsune_p2p/timestamp/src/lib.rs

```diff
@@ -15,6 +15,13 @@ use std::convert::{TryFrom, TryInto};
 
 pub use crate::error::{TimestampError, TimestampResult};
 
+/// This represents general compatibility between kitsune versions.
+/// Instances are compatible if and only if they share the same protocol version.
+///
+/// If there is a breaking change at any level of the wire protocol, including
+/// changes to serialization, this version should be incremented.
+pub const KITSUNE_PROTOCOL_VERSION: u32 = 0;
```

Comment:
wut?--why is this in the timestamp crate? maybe deserves an explanation in the docs

Reply:
Yeah it does seem out of place, doesn't it? It's because kitsune_p2p_timestamp has the honor of being "most upstream kitsune crate". A change in the timestamp format would also necessitate a bump to the protocol version so it kinda makes sense. Pragmatically, `holochain_zome_types` depends on `kitsune_p2p_timestamp` but not `kitsune_p2p`, so that's why this move was actually needed in this instance. But yeah, I don't know what a better solution would be.